### ISSUE TITLE
Invalid color space error when doing equal partitioning

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#394]
When I run this Regent code:

```
import 'regent'
local struct face {
  x : int
}
local task foo(faces : region(ispace(int3d), face)) end
local task main()
  var grid_x = ispace(int3d, {5, 1, 1})
  var x_faces = region(grid_x, face)
  var tiles_private  = ispace(int3d, {4, 1, 1})
  var p_x_faces_equal = partition(equal, x_faces, tiles_private)
  for c in tiles_private do
    foo(p_x_faces_equal[c])
  end
end
regentlib.start(main)
```

I get the following error:

```
[0 - 7fd5e8bcd700] {5}{runtime}: [error 39] LEGION ERROR: Invalid color space color for child 3 of partition 1 (from file /home/manolis/sw/legion/runtime/legion/region_tree.cc:6077)[error 39] 
For more information see:
http://legion.stanford.edu/messages/error_code.html#error_code_39
```

Edit: removed an irrelevant line

**Comment by lightsighter** [comment]
So the strange thing that I see when I run this example is the compiler is generating a call to 'legion_index_partition_create_blockify_3d' and when it does the blockify object looks like this:
```
(gdb) p blockify_
$13 = {
  block_size = {
    x = {2, 1, 1}
  }, 
  offset = {
    x = {0, 0, 0}
  }
}
```
So the color space is only going to be (3x1x1) instead of (4x1x1). If the compiler wants to guarantee that the color space is the same as the one the user specified then it should use create_partition_by_restriction instead. In the case of partition by blockify the runtime will only make the partition with a color space as big as is necessary.

**Comment by elliottslaughter** [comment]
@lightsighter Does equal partition not do what we want now? I remember we set it up this way because before deppart blockify was the only reasonable way to get the kinds of shapes we wanted in the multi-dimensional case. But if equal actually works now we should just do that.

**Comment by lightsighter** [comment]
It looks like Realm has implemented create_equal_partition for an arbitrary number of dimensions now but my guess is that it doesn't do what you want yet:

https://gitlab.com/StanfordLegion/legion/blob/master/runtime/realm/deppart/partitions.cc#L131-145

I think the right thing to do here is for the compiler to start calling create_partition_by_restriction so that it can align the color space with the blocking scheme that it wants. Right now the C API is providing a translation from the deprecated blockify objects to the runtime's convenience create_partition_by_blockify (which doesn't take a color space argument, so the runtime get's the convenience method makes its own). If the compiler also wants to specify the color space then it should really be calling create_partition_by_restriction directly.

**Comment by elliottslaughter** [comment]
@manopapad This may be fixed in 5ce8eda, can you test? Thanks.

**Comment by manopapad** [comment]
Fixed!
